TMTumblrSDK/Core/TMSDKFunctions.m

```diff
@@ -58,21 +58,29 @@ @implementation TMSDKFunctions
 NSString *TMDictionaryToQueryString(NSDictionary *dictionary) {
     NSMutableArray *parameters = [NSMutableArray array];
     
-    void (^addParameter)(NSString *key, NSString *value) = ^(NSString *key, NSString *value) {
-        [parameters addObject:[NSString stringWithFormat:@"%@=%@", TMURLEncode(key), TMURLEncode(value)]];
-    };
-    
     for (NSString *key in [[dictionary allKeys] sortedArrayUsingSelector:@selector(caseInsensitiveCompare:)]) {
-        id value = dictionary[key];
-        
-        if ([value isKindOfClass:[NSArray class]]) {
-            for (NSString *arrayValue in (NSArray *)value)
-                addParameter(key, arrayValue);
-        } else
-            addParameter(key, value);
+        TMAddKeyValuePairToQueryStringMutableArray(key, dictionary[key], parameters);
     }
     
     return [parameters componentsJoinedByString:@"&"];
 }
 
+#pragma mark - private
+
+void TMAddKeyValuePairToQueryStringMutableArray(NSString *key, id value, NSMutableArray *parameters) {
+    if ([value isKindOfClass:[NSDictionary class]]) {
+        for (NSString *subKey in [((NSDictionary *)value).allKeys sortedArrayUsingSelector:@selector(caseInsensitiveCompare:)]) {
```

Comment:
Maybe `NSDictionary *dictionary = (NSDictionary *)value;` on the previous line, since `value` is used twice as a dictionary?